### PR TITLE
Feature/16612 pupil content changes

### PR DESCRIPTION
**Pull request by jon-shipley**
Copy changes

But..

* api change on the spa client login / questions to return date of birth
* refactored pupil-authentication.service to move data access to data services
* jwtToken createToken() signature change
* add new warmup loading component to SPA as the html text is different for warmup questions
* and finally the copy changes requested
* and updated cucumber tests

### CODE REVIEW

**Comment by jon-shipley** [comment]
@activemq @sinhaamit1980 - please can you approve the cucumber tests I've altered as per the copy changes requested in 16612

As this deals with the SPA authentication probably best if we get 2 approvals or more.

File: admin/services/data-access/pupil.data.service.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+/**
+ * Update a single object with the new fields from `doc`
+ * @param {string} id Object._id
+ * @param {Object} doc Document to pass to mongo
+ * @return {Promise}
+ */
+pupilDataService.update = async function (id, doc) {
+  return new Promise((resolve, reject) => {
+    Pupil.updateOne({_id: id}, doc, (error) => {
```
Comment (GeorgeChatzigiannis) [inline admin/services/data-access/pupil.data.service.js]: I am surprised this works with a callback for errors.
isn't there a promisified version of this?

Comment (GeorgeChatzigiannis) [reply admin/services/data-access/pupil.data.service.js]: you could also promisify it with bluebird I suppose and it would be the same result(?)

Comment (jon-shipley) [reply admin/services/data-access/pupil.data.service.js]: Not that I know of.

You can put `.exec()` on the end to get a Promise, but then it does not wait for the response from the server.  So not the same thing.

I don't see why you expect it not to work.  Care to elaborate?

Comment (GeorgeChatzigiannis) [reply admin/services/data-access/pupil.data.service.js]: I am expecting it to work don't get me wrong but I was also expecting it to work under a promise just fine and wait for the server.

Comment (jon-shipley) [reply admin/services/data-access/pupil.data.service.js]: The underlying mongo api is callback-only. I suspect `updateOne`  is a minimal wrapper around it.

File: admin/services/data-access/pupil.data.service.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+'use strict'
```
Comment (GeorgeChatzigiannis) [inline admin/services/data-access/pupil.data.service.js]: oh boy we are gonna have conflicts on this one, gonna be fun!!!

File: admin/services/data-access/pupil.data.service.js (hunk 1 above)
Comment (pris54) [reply admin/services/data-access/pupil.data.service.js]: Mongoose updateOne returns a query:
http://mongoosejs.com/docs/api.html#model_Model.updateOne

Comment (GuyHarwood) [reply admin/services/data-access/pupil.data.service.js]: the official mongoClient is promisified and much cleaner.  But then we lose the schemas...

Comment (jon-shipley) [reply admin/services/data-access/pupil.data.service.js]: @pris54 - I'm not sure from your comment if there is a better way to handle this or not.  Can you clarify please?

**Pull request merged**